authz/providers/azure/rbac/rbac.go

```
@@ -0,0 +1,247 @@
+		glog.V(10).Infof("binary data:%s", binaryData)
+	}
+
+	req, err := http.NewRequest(http.MethodPost, checkAccessURL.String(), buf)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating check access request")
+	}
+	// Set the auth headers for the request
+	req.Header = a.headers
+
+	if glog.V(10) {
+		cmd, _ := http2curl.GetCurlCommand(req)
+		glog.V(10).Infoln(cmd)
+	}
+
+	resp, err := a.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "error in check access request execution")
+	}
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "error in reading response body")
+	}
+
+	glog.V(10).Infof("checkaccess response: %s", string(data))
+	defer resp.Body.Close()
+	glog.V(10).Infof("Configured ARM instance: %d", a.armCallLimit)
+	if resp.StatusCode != http.StatusOK {
+		glog.Errorf("error in check access response. error code: %d, response: %s", resp.StatusCode, data)
+		if resp.StatusCode == http.StatusTooManyRequests {
+			glog.V(10).Infoln("Moving to another ARM instance!")
+			a.client.CloseIdleConnections()
+			// add prom metrics for this scenario
+		}
+		return nil, errors.Errorf("request %s failed with status code: %d and response: %s", req.URL.Path, resp.StatusCode, string(data))
+	} else {
+		remaining := resp.Header.Get(remaingSubReadARMHeader)
+		glog.Infof("Remaining request count in ARM instance:%s", remaining)
+		count, _ := strconv.Atoi(remaining)
+		if count < a.armCallLimit {
+			if glog.V(10) {
+				glog.V(10).Infoln("Moving to another ARM instance!")
+			}
+			// Usually ARM connections are cached by destinatio ip and port
+			// By closing the idle connection, a new request will use different port which
+			// will connect to different ARM instance of the region to ensure there is no ARM throttling
+			a.client.CloseIdleConnections()
+		}
+	}
+
+	// Decode response and prepare k8s response
+	response, err := ConvertCheckAccessResponse(data)
```

why the err is not checked before writing to cache?

---

it was a miss. Handled err case. Thanks!